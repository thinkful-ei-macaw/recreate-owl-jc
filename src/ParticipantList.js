import React from 'react'
import Participant from './Participant';

const ParticipantList = (props) => {

return (
  <div>
    {props.participants.map(participant => (participant.inSession
    ? <Participant 
    key={participant.id}
    avatar={participant.avatar}
    name={participant.name}
    onStage={participant.onStage}/>
    : null))}
  </div>
);
}

export default ParticipantList