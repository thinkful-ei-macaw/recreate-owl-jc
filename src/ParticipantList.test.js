import React from 'react';
import ReactDOM from 'react-dom';
import ParticipantList from './ParticipantList';
import renderer from 'react-test-renderer';

describe('ParticipantList Component Tests', () => {

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<ParticipantList participants={[
      {
          id: 1,
          name: 'Koren Templeton',
          avatar:
              'https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1',
          inSession: true,
          onStage: true
      },
      {
          id: 2,
          name: 'Caty Flucker',
          avatar:
              'https://robohash.org/doloredolorescupiditate.jpg?size=200x200&set=set1',
          inSession: true,
          onStage: false
      }]} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the list with info', () => {
    const tree = renderer
    .create(<ParticipantList participants={[
      {
          id: 1,
          name: 'Koren Templeton',
          avatar:
              'https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1',
          inSession: true,
          onStage: true
      },
      {
          id: 2,
          name: 'Caty Flucker',
          avatar:
              'https://robohash.org/doloredolorescupiditate.jpg?size=200x200&set=set1',
          inSession: true,
          onStage: false
      }]} />)
      .toJSON();
      expect(tree).toMatchSnapshot();
  })

});
