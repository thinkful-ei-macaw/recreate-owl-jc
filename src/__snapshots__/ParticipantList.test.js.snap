// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ParticipantList Component Tests renders the list with info 1`] = `
<div>
  <div>
    <img
      alt="Avatar"
      className="participant participant-img"
      src="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
    />
    <p>
      Koren Templeton
    </p>
    <p>
      on stage
    </p>
  </div>
  <div>
    <img
      alt="Avatar"
      className="participant participant-img"
      src="https://robohash.org/doloredolorescupiditate.jpg?size=200x200&set=set1"
    />
    <p>
      Caty Flucker
    </p>
  </div>
</div>
`;
